Death Dotter Plugin

dmgear opened the pull request:
This is a simple plugin that compares the radii of two players, and if the distance between two players falls below the threshold that is the sum of their radii, we hide the local client player similar to entity hider.

The purpose of this plugin is to perform the same function as entity hider + player outline in pvp, but without having to play with an invisible character.

tylerwgrass commented:
Please do not close/reopen your PR. If there are build issues, you can make a commit to fix and push it to this PR branch.

dmgear commented:
> Please do not close/reopen your PR. If there are build issues, you can make a commit to fix and push it to this PR branch.

Understood, my apologies. Thank you.

cdfisher commented:
Not a thorough review, and both of these are fairly minor but:
* You can just remove `enabledByDefault = true` entirely since that's the default behavior
* It doesn't look like your config key actually gets used anywhere so you can probably just remove the config altogether.

dmgear commented:
> Not a thorough review, and both of these are fairly minor but:
> 
> * You can just remove `enabledByDefault = true` entirely since that's the default behavior
> * It doesn't look like your config key actually gets used anywhere so you can probably just remove the config altogether.

Thank you for your feedback. I will remove `enabledByDefault = true`. However, the config key is being used in the updateConfig() method in my plugin class.

cdfisher commented:
> However, the config key is being used in the updateConfig() method in my plugin class.

The only thing you're doing with `updateConfig()` is updating the config value which you don't actually use anywhere. Your config, `updateConfig()`, `onConfigChanged()`, and all the references to them could all be removed while having no effect on your plugin because none of them are used in any meaningful way.

dmgear commented:
> > Not a thorough review, and both of these are fairly minor but:
> > 
> > * You can just remove `enabledByDefault = true` entirely since that's the default behavior
> > * It doesn't look like your config key actually gets used anywhere so you can probably just remove the config altogether.
> 
> Thank you for your feedback. I will remove `enabledByDefault = true`. However, the config key is being used in the updateConfig() method in my plugin class.

Ah! I see what you mean now! I am new to runelite development so please bear with me. To my understanding, to register the plugin with runelite to get it to show up in the client I must use the config. I notice now that if I remove the config and its associated methods my plugin no longer shows up in the runelite client. What should I do instead?

dmgear commented:
> > However, the config key is being used in the updateConfig() method in my plugin class.
> 
> The only thing you're doing with `updateConfig()` is updating the config value which you don't actually use anywhere. Your config, `updateConfig()`, `onConfigChanged()`, and all the references to them could all be removed while having no effect on your plugin because none of them are used in any meaningful way.

I understand. I was misunderstanding the purpose of the config class. I have removed my config and also refactored my plugin.

tylerwgrass commented:
You have added a bunch of files that shouldn't be here.
You should only be adding your `plugins/deathdotter` file.

dmgear commented:
> You have added a bunch of files that shouldn't be here. You should only be adding your `plugins/deathdotter` file.

Unsure what happened, I didn't make changes in any of those files. I unstaged those changes by resetting my commit. It asked me to do a pull before I pushed again, and I pushed just now. But I think it committed a bunch of extra files again.

dmgear commented:
On this commit: "https://github.com/runelite/plugin-hub/pull/7683/commits/b29e66e9ab32e3d6c798ca31351269e363411795"

You can see I just pushed my plugin. But then here: "https://github.com/runelite/plugin-hub/pull/7683/commits/086cacaa367b5997a0bb69fc844e1019a51d651c"

There's a bunch of extra files from the merge. Please advise.

tylerwgrass commented:
Delete the files and push a new commit instead of resetting.

dmgear commented:
Done.

Felanbird commented:
(it's not done)

dmgear commented:
Reset to previous commit. I think this is good now?
